src/components/patterns/algolia-course-list.tsx

```diff
@@ -0,0 +1,127 @@
+"use client";
+
+import algoliasearch from "algoliasearch/lite";
+import {useHits,usePagination, Configure} from "react-instantsearch";
+import {InstantSearchNext} from "react-instantsearch-nextjs";
+import {useEffect, useMemo, useState} from "react";
+import {Hit as HitType} from "instantsearch.js";
+import SummerCourse from "@components/algolia-results/summer-course/summer-course";
+import useFavorites from "@lib/hooks/useFavorites";
+import { useSearchParams } from "next/navigation";
+import { AlgoliaHit } from "@components/algolia-results/default";
+
+type Props = {
+  appId: string
+  searchIndex: string
+  searchApiKey: string
+}
+
+type FavoriteItem = {
+  uuid: string;
+  title: string;
+  path?: string; 
+  units: number;
+}
+
+type FavoritesList = FavoriteItem[];
+
+const AlgoliaCourseList = ({appId, searchIndex, searchApiKey}: Props) => {
+  const searchClient = useMemo(() => algoliasearch(appId, searchApiKey), [appId, searchApiKey])
+  const [urlParams, setUrlParams] = useState<FavoritesList>([]);
+  const { favs, addFav } = useFavorites()
+  const searchParams = useSearchParams();
+
+  useEffect(() => {
+    const favParamData = searchParams.get("fav");
```

Comment:
The functionality to check for params and display the list of favorites is handled here. I included the uuid, title, and units in the params since we need that extra information for the Favorites List display. Thoughts? 😅

Reply:
I don't believe we need anything other than the `uuid` in the parameters. Theres 2 scenarios I think here
1. I am a prospective student.
    1. I choose some favorite courses
    2. I land on this component
    3. I see only my favorite items
4. I am a parent/friend
    1. My child/friend shares a link
    2. I land on this component with the uuids
    3. I can view the favorite items as just a plain list. Nothing special. 

So I don't think this `useEffect` is necessary. I think you should be able to just use `searchParams.get('favs')?.split(',')` below